### PR TITLE
feat: add support for custom routes

### PR DESCRIPTION
**Pull request by eduardoboucas**
#### Summary

Reads the `path` in-source configuration property and writes the resulting routes into a `routes` field in the manifest. It only does so for v2 functions.

Part of https://github.com/netlify/pod-dev-foundations/issues/572.

### CODE REVIEW

File: src/utils/urlpattern.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { URLPattern } from 'urlpattern-polyfill'
```
Comment (eduardoboucas) [inline src/utils/urlpattern.ts:1]: Taken from https://github.com/netlify/edge-bundler/blob/main/node/utils/urlpattern.ts.

File: tests/fixtures-esm/v2-api-with-path/with-named-group.mjs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export default async () =>
+  new Response('<h1>Hello world</h1>', {
+    headers: {
+      'content-type': 'text/html',
+    },
+  })
+
+export const config = {
+  path: '/products/:id',
```
Comment (lukasholzer) [inline tests/fixtures-esm/v2-api-with-path/with-named-group.mjs:9]: do we want to add a test for multiple named groups as well?

like `/products/:category/:variant/:sku`

**Review by lukasholzer — APPROVED** [review]
Nothing blocking but I would add additional tests for the `path` property covering some more scenarios like:

- invalid url patterns 
- paths starting without a slash
- longer nested URLs
- deep nested URLs like `/product/:id/view`

File: tests/fixtures-esm/v2-api-with-path/with-named-group.mjs (hunk 2 above)
Comment (eduardoboucas) [reply tests/fixtures-esm/v2-api-with-path/with-named-group.mjs:9]: Great idea. Done in https://github.com/netlify/zip-it-and-ship-it/pull/1523/commits/b7fbde318debce338ca3a4a1193ece0226ca6271.

**Review by lukasholzer — APPROVED** [review]
Awesome!

**Pull request merged**